ksqldb-engine/src/main/java/io/confluent/ksql/util/ValidationSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -15,192 +15,103 @@
 import io.confluent.ksql.query.KafkaStreamsBuilder;
-import io.confluent.ksql.query.QueryError;
 import io.confluent.ksql.query.QueryErrorClassifier;
 import io.confluent.ksql.query.QueryId;
-import io.confluent.ksql.rest.entity.StreamsTaskMetadata;
+import java.time.Duration;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
 import java.util.Map;
-import java.util.Set;
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.stream.Collectors;
-import org.apache.kafka.streams.KafkaStreams;
-import org.apache.kafka.streams.LagInfo;
 import org.apache.kafka.streams.StreamsConfig;
-import org.apache.kafka.streams.StreamsMetadata;
-import org.apache.kafka.streams.errors.StreamsException;
 import org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler;
-import org.apache.kafka.streams.processor.internals.namedtopology.KafkaStreamsNamedTopologyWrapper;
+import org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class ValidationSharedKafkaStreamsRuntimeImpl implements SharedKafkaStreamsRuntime {
-  private static final Logger LOG = LoggerFactory.getLogger(SharedKafkaStreamsRuntimeImpl.class);
-
-  private final KafkaStreamsNamedTopologyWrapper kafkaStreams;
-  private final Map<String, Object> streamsProperties;
-  private final Map<String, PersistentQueriesInSharedRuntimesImpl> metadata;
-  private final Map<QueryId, Set<SourceName>> sources;
-  private final KafkaStreamsBuilder kafkaStreamsBuilder;
+public class ValidationSharedKafkaStreamsRuntimeImpl extends SharedKafkaStreamsRuntime {
+  private final Logger log = LoggerFactory.getLogger(ValidationSharedKafkaStreamsRuntimeImpl.class);
 
   public ValidationSharedKafkaStreamsRuntimeImpl(
-      final SharedKafkaStreamsRuntime sharedKafkaStreamsRuntime
+      final SharedKafkaStreamsRuntime sharedRuntime
   ) {
-    streamsProperties = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getStreamProperties());
-    streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG,
-        "_confluent-ksql-" + UUID.randomUUID() + "-validation"
+    super(
+        sharedRuntime.getKafkaStreamsBuilder(),
+        getSandboxStreamsProperties(sharedRuntime),
+        new QueryMetadataImpl.TimeBoundedQueue(Duration.ofHours(1), 0)
     );
-    kafkaStreamsBuilder = sharedKafkaStreamsRuntime.getKafkaStreamsBuilder();
-    kafkaStreams = kafkaStreamsBuilder.buildNamedTopologyWrapper(streamsProperties);
-    metadata = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getMetadata());
-    sources = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getSourcesMap());
-    for (PersistentQueriesInSharedRuntimesImpl queryMetadata : metadata.values()) {
+
+    for (BinPackedPersistentQueryMetadata query : sharedRuntime.collocatedQueries.values()) {
       //kafkaStreams.addNamedTopology(queryMetadata.getTopology());
```

Comment:
This was commented out because adding named topologies  was breaking the tests. I think it might have been because we were reusing the topology. We need to be able to copy the topology. Do we have this work tracked?

Reply:
This is covered in my next PR, which includes all the work related to the query validation and sandboxed runtimes that I'm splitting out of the initial mega-PR

So short answer is, yes. Ticket is https://confluentinc.atlassian.net/browse/KCI-909